langchain/chains/query_constructor/ir.py

```diff
@@ -75,8 +75,8 @@ class Comparator(str, Enum):
     GTE = "gte"
     LT = "lt"
     LTE = "lte"
-    CONTAIN = "contain"
-    LIKE = "like"
+    CONTAIN = "list_contain"
+    LIKE = "string_pattern_like"
```

Comment:
this name still feels ambiguous to me, what is the LIKE comparison doing again? is it a regex match?

Reply:
Personally I think this function should be bounded with datatypes. Unfortunately I didn't find any place to insert the allowed datatypes with those functions. Do you have better idea?

We also explored some new ways to implement [self-query retriever](https://docs.myscale.com/en/sample-applications/openai-function-call/) with openai's function call. We found pretrained LLMs are good SQL translator so it  is probably a good new to SQL based vector databases.

Reply:
> 

Sorry I probably misunderstand the issue. Do you mean the `LIKE` is ambiguous? Perhaps `SUBSTRING_MATCH` would be a better fit?